backends/dpdk/dpdkArch.cpp

```diff
@@ -727,11 +727,15 @@ const IR::Node *InjectJumboStruct::preorder(IR::Type_Struct *s) {
     return s;
 }
 
-const IR::Node *InjectOutputPortMetadataField::preorder(IR::Type_Struct *s) {
-    if (structure->isPNA() && s->name.name == structure->local_metadata_type) {
-        s->fields.push_back(new IR::StructField(
-            IR::ID(PnaMainOutputMetadataOutputPortName), IR::Type_Bits::get(32)));
-        LOG3("Metadata structure after injecting output port:" << std::endl << s);
+const IR::Node *InjectFixedMetadataField::preorder(IR::Type_Struct *s) {
+    if  (s->name.name == structure->local_metadata_type) {
```

Comment:
can compare originalName instead of name?

Reply:
In many cases, originalName is also modified by the frontend. Have faced issues in the passt. We rely on the internal name which is present as name.name.